flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveSourceDynamicFileEnumerator.java

```diff
@@ -159,6 +161,9 @@ RowData createRowData(RowType rowType, Map<String, String> partitionSpec) {
                     throw new UnsupportedOperationException(
                             "Unsupported type for dynamic filtering:" + rowType.getTypeAt(i));
             }
+            if (!fieldType.isNullable() && convertedValue == null) {
```

Comment:
What if the field type is nullable ?

Reply:
If the filed is nullable, the null value will set to the field and compare with the DynamicFilteringData normally, which will not cause any exception. The case has been covered in the original test cases.

Reply:
What if the filed type is declared as notnullable, but the actual data is in here is null?
Seems we will miss such partition and then cause lossing data.
Is it possible to handle such case? Or at least we should throw an exception.

Reply:
The type is acquired from the DynamicFilteringData, which is a subset of the row type of the dim side. Since the DynamicFilteringData is built with the data from dim side, no null values should exist in a non-null field in the DynamicFilteringData.
While a null value presents here probably because the partition name is DEFAULT_PARTITION_NAME for a non-string field. The null value is impossible to match the DynamicFilteringData if the field is non-null, so it should be safe to skip the matching and discard the partition immediately.

Reply:
make sense